userspace/libscap/ringbuffer/devset.c

```diff
@@ -38,3 +38,38 @@ int32_t devset_init(struct scap_device_set *devset, size_t num_devs, char *laste
 
 	return SCAP_SUCCESS;
 }
+
+void devset_free(struct scap_device_set *devset)
+{
+	if(devset == NULL || devset->m_devs == NULL)
+	{
+		return;
+	}
+
+	uint32_t j;
+	for(j = 0; j < devset->m_ndevs; j++)
+	{
+		struct scap_device *dev = &devset->m_devs[j];
+		if(dev->m_buffer != MAP_FAILED)
+		{
+#ifdef _DEBUG
+			int ret;
+			ret = munmap(dev->m_buffer, dev->m_buffer_size);
+#else
+			munmap(dev->m_buffer, dev->m_buffer_size);
+#endif
+			ASSERT(ret == 0);
+		}
+
+		if(dev->m_bufinfo != MAP_FAILED)
+		{
+			munmap(dev->m_bufinfo, dev->m_bufinfo_size);
+		}
```

Comment:
I'll double check when I get a minute, but I think `devset_init` initializes `m_bufinfo` to `MAP_FAILED` [here](https://github.com/falcosecurity/libs/blob/739557dfaeaa65f13084cc1a142895a02ef27906/userspace/libscap/ringbuffer/devset.c#L29), because it is never used again with eBPF, it should always retain that value and skip the code block (at least that is what I remember off the top of my head 😅 )

Reply:
Oh, thanks, I've missed it in the BPF probe, so it should work fine!